core/src/wasm/overrides.rs

```diff
@@ -0,0 +1,340 @@
+
+static INITIALIZED: AtomicBool = AtomicBool::new(false);
+static DISABLED: AtomicBool = AtomicBool::new(false);
+
+pub fn disable_persistent_timer_overrides() {
+    DISABLED.store(true, Ordering::Relaxed);
+}
+
+pub fn init_timer_overrides() -> Result<(), String> {
+    // If already initialized exit
+    if INITIALIZED.load(Ordering::Relaxed) {
+        return Ok(());
+    } else {
+        INITIALIZED.store(true, Ordering::Relaxed);
+    }
+
+    // If disabled exit
+    if DISABLED.load(Ordering::Relaxed) {
+        return Ok(());
+    }
+
+    // Persistent timers shall only be injected in a web context.
+    if !is_web() {
+        return Ok(());
+    }
+
+    // Get the global window object
+    let window = window().unwrap();
+
+    if Reflect::get(&window, &JsValue::from_str("__persistent_timers")).is_err() {
+        // This means __persistent_timers is not yet set, so we set it.
+        let _ = Reflect::set(
+            &window,
+            &JsValue::from_str("__persistent_timers"),
+            &JsValue::from(true),
+        )
+        .map_err(|err| err.as_string());
+    } else {
+        // Do not allow initializing persistent timers if already set.
+        return Ok(());
+    }
+
+    // Create a new TimerManager and wrap it in an Rc<RefCell<_>> so
+    // that it can be shared across functions and still share mutable state.
+    let timer_manager = Rc::new(RefCell::new(TimerManager::new()));
+
+    // Wrap the setInterval function so it returns the timer ID
+    let set_interval_closure = {
+        let timer_manager = timer_manager.clone();
+        Closure::wrap(Box::new(move |callback: JsValue, time: u32| -> JsValue {
+            let id = timer_manager.borrow_mut().set_interval(&callback, time);
+            JsValue::from_f64(id as f64) // Convert the ID to JsValue and return
+        }) as Box<dyn FnMut(JsValue, u32) -> JsValue>)
+    };
+
+    // Wrap the clearInterval function
+    let clear_interval_closure = {
+        let timer_manager = timer_manager.clone();
+        Closure::wrap(Box::new(move |id: u32| {
+            timer_manager.borrow_mut().clear_interval(id);
+        }) as Box<dyn FnMut(u32)>)
+    };
+
+    // Wrap the setTimeout function so it returns the timer ID
+    let set_timeout_closure = {
+        let timer_manager = timer_manager.clone();
+        Closure::wrap(Box::new(move |callback: JsValue, time: u32| -> JsValue {
+            let id = timer_manager.borrow_mut().set_timeout(&callback, time);
+            JsValue::from_f64(id as f64) // Convert the ID to JsValue and return
+        }) as Box<dyn FnMut(JsValue, u32) -> JsValue>)
+    };
+
+    // Wrap the clearTimeout function
+    let clear_timeout_closure = {
+        let timer_manager = timer_manager.clone();
+        Closure::wrap(Box::new(move |id: u32| {
+            timer_manager.borrow_mut().clear_timeout(id);
+        }) as Box<dyn FnMut(u32)>)
+    };
+
+    // Set the new functions on the window object
+    js_sys::Reflect::set(
```

Comment:
pair these with declarations - make closure, override, make closure, override...  

(as opposed to make 4 closures and override all of them)...

makes code grouped better (IMO)

Reply:
a2f25c0efe6a1120867a044ce3e95715ed5d9804